pkg/scm/scm.go

```diff
@@ -13,22 +15,52 @@ import (
 
 // DownloaderForSource determines what SCM plugin should be used for downloading
 // the sources from the repository.
-func DownloaderForSource(s string) build.Downloader {
-	// If the source starts with file:// and there is no GIT binary, use 'file'
-	// SCM plugin
-	if (strings.HasPrefix(s, "file://") || strings.HasPrefix(s, "/")) && !hasGitBinary() {
-		return &file.File{util.NewFileSystem()}
+func DownloaderForSource(s string) (build.Downloader, string, error) {
+	// If the source is using file:// protocol but it is not a GIT repository,
+	// trim the prefix and treat it as a file copy.
+	if strings.HasPrefix(s, "file://") && !isLocalGitRepository(s) {
+		s = strings.TrimPrefix(s, "file://")
 	}
 
-	g := git.New()
-	if g.ValidCloneSpec(s) {
-		return &git.Clone{g, util.NewFileSystem()}
+	// If the source is file:// protocol and it is GIT repository, but we don't
+	// have GIT binary to fetch it, treat it as file copy.
+	if strings.HasPrefix(s, "file://") && !hasGitBinary() {
+		s = strings.TrimPrefix(s, "file://")
 	}
 
-	glog.Errorf("No downloader defined for %q source URL", s)
-	return nil
+	// If the source is valid GIT protocol (file://, git://, git@, etc..) use GIT
+	// binary to download the sources
+	if g := git.New(); g.ValidCloneSpec(s) {
```

Comment:
if you specify git:// or git@ protocol and you don't have binary, then you will get an error when attempting to clone.